saleor/giftcard/utils.py

```diff
@@ -165,6 +165,7 @@ def gift_cards_create(
     non_shippable_gift_cards = []
     expiry_date = calculate_expiry_date(settings)
     for order_line in gift_card_lines:
+        fulfillment_line = order_line.fulfillment_lines.last()
```

Comment:
maybe we should fetch related lines once rather than calling these .last()s on each iteration? I'm assuming this are not cached by django and create extra queries, but that would have to be checked.
Quick solution would be, before the loop:
```
fulfillment_lines = FulfillmentLine.objects.filter(order_line_pk__in=gift_card_lines).order_by("-pk")
```
and inside something like
```
fulfillment_line = next(filter(lambda f: f.order_line_id == order_line.id, fulfillment_lines), None)
```
On the other hand maybe thats not an issue at all, seems unlikely that user will ever buy lots of these at once, right?

Reply:
Yes, I think that in most cases it will be only one gift card line. But maybe it's a good idea to introduce a solution with a constant query amount.